src/server.js

```diff
@@ -107,13 +107,10 @@ function setupRenderer() {
         console.error('ROUTER ERROR:', pretty.render(error));
         res.status(500);
         hydrateOnClient();
-      } else if (!renderProps) {
-        res.status(500);
-        hydrateOnClient();
-      } else {
+      } else if (renderProps) {
         const component = (
           <Provider store={store} key="provider">
-            <RoutingContext />
+            <RoutingContext {...renderProps} />
```

Comment:
@dts let's bring the 404 handling in in a different PR post RR2. i'll keep this PR open for now.